src/test/unit/test_lumigo_utils.py

```diff
@@ -445,25 +447,32 @@ def test_get_edge_host(arg, host, monkeypatch):
     assert get_edge_host("region") == host
 
 
-def test_report_json_extension(monkeypatch, reporter_mock):
+def test_report_json_extension_spans_mode(monkeypatch, reporter_mock):
     monkeypatch.setattr(Configuration, "should_report", True)
     monkeypatch.setenv("LUMIGO_USE_TRACER_EXTENSION", "TRUE")
-    single = [{"a": "b"}]
-    to_send = _create_request_body(single, True).encode()
-    md5str = str(hashlib.md5(to_send + b"#DONE#").hexdigest())
-    file_name = f"{md5str}_single"
+    spans = []
+    size_factor = 100
+    for i in range(size_factor):
+        spans.append(
+            {
+                i: "a" * size_factor,
+            }
+        )
+    report_json(None, spans)
+
+    files_paths = []
+    for span in spans:
+        to_send = aws_dump(span).encode() + b"#DONE#"
+        file_name = f"{hashlib.md5(to_send).hexdigest()}_span"
+        file_path = f"/tmp/lumigo-spans/{file_name}"
+        files_paths.append(file_path)
+    done_object = {"spansCount": len(spans)}
+    done_span = aws_dump(done_object).encode() + b"#DONE#"
+    file_name = f"{hashlib.md5(done_span).hexdigest()}_done"
     file_path = f"/tmp/lumigo-spans/{file_name}"
-    asserting_extension(file_path, single)
-    # test that same file doesnt cause error
-    asserting_extension(file_path, single)
-
-
-def asserting_extension(file_path, single):
-    duration = report_json(None, [{"a": "b"}])
-    content = open(file_path, "r").read()[:-6]
-    span_from_file = json.loads(content)
-    assert span_from_file == single
-    assert duration == 0
+    files_paths.append(file_path)
+    for file_name in glob.glob("/tmp/lumigo-spans/*"):
+        assert file_name in files_paths
```

Comment:
it seems like this test mainly check the file name building. 
Is that what you want to do?
maybe we can read the data and check that this is a son and that it ends with "DONE"?